### PR TITLE
2FA SMS recovery

### PR DESCRIPTION
**Pull request by gudmundur**
Adds SMS recovery steps to account where 2FA is enabled with a sms recovery number.

Uses https://github.com/heroku/api/pull/3410, but should be safe to merge before.

### What's missing

* [x] Handle errors from `/account/sms/recovery`
* [x] Only show SMS recovery link if `sms_number` present
* [x] Display partial `sms_number` in SMS recovery step. ie. "Number ending with 1234"
* [x] Fixing tests & adding for where it applies

Second factor page:

screen shot 2015-01-15 at 17 03 34

Recovery page:

screen shot 2015-01-15 at 17 03 41

Second factor page if no `sms_number` set on account:

screen shot 2015-01-16 at 14 03 35

Recovery page if no `sms_number` set on account:

screen shot 2015-01-16 at 14 33 11

### CODE REVIEW

**Comment by brandur** [comment]
@gudmundur @dizzyup Slick! The code looks good to me, and thanks for adding comprehensive tests as well. I assume the page design has been vetted pretty well already?

**Comment by markpundsack** [comment]
@brandur The design was vetted on a Dashboard issue (heroku/dashboard-v6#853).

**Comment by markpundsack** [comment]
@gudmundur @appleton Would be nice to have screenshots here for posterity, including the 2 variations (with or without SMS enabled).

**Comment by brandur** [comment]
@markpundsack Ah nice, thank-you.

**Comment by appleton** [comment]
@markpundsack added to the initial description.

**Comment by gudmundur** [comment]
@brandur Added aa624a7 to fix a "Get a SMS code" link on the recovery page, even if no number was set.

Discovered this issue while testing this change against a non-SMS recovery API.

**Comment by gudmundur** [comment]
@markpundsack @brandur Added screenshots of what the flow looks like without `sms_number` set. This is what it will all look like without support in API, aka. this PR is safe to merge before API change is merged.

**Comment by gudmundur** [comment]
@brandur maybe wait with merging this if we're renaming the endpoints.

**Comment by brandur** [comment]
> @brandur maybe wait with merging this if we're renaming the endpoints.

+1. Thank-you.

**Comment by gudmundur** [comment]
@brandur Updated to reflect changes in the API PR.

File: lib/identity/design.rb [hunk 1]
```diff
@@ -29,6 +29,14 @@ class Design < Sinatra::Base
         slim :"two-factor", layout: @layout
       end
 
+      get "/two-factor/recovery" do
+        slim :"account/two-factor/recovery", layout: @layout
```
Comment (brandur) [inline lib/identity/design.rb]: Sorry for the bikeshed, but mind explicitly setting the SMS number here?

``` ruby
@sms_number = fetch_sms_number
```

Or something like that?

**Comment by brandur** [comment]
Great job guys! Thanks for adding in all the really nitpicky changes :/

**Pull request merged**